liquidprompt

```diff
@@ -2571,6 +2573,46 @@ _lp_time_color() {
     lp_time_color="${LP_COLOR_TIME}${_LP_TIME_SYMBOL}${NO_COL}"
 }
 
+#######################
+# Container detection #
+#######################
+_lp_container() {
+    (( LP_ENABLE_CONTAINER )) ||  return 2
+
+    local container=""
+
+    if [[ -n "${SINGULARITY_CONTAINER-}" ]] ||
+        [[ -n "${SINGULARITY_NAME-}" ]]; then
+        container="Sing"
+
+    elif [[ -e /run/.containerenv ]]; then
+        container="Pdmn"
+
+    elif [[ -e /proc/self/cgroup ]]; then
+        if  grep -q "docker" /proc/self/cgroup ; then
+	    container="Dokr"
+
+        elif  grep -q "lxc" /proc/self/cgroup ; then
+              container="LXC"
+        fi
+    fi
+
+
+    if [[ -n "${container}" ]] ; then
+       lp_container=${container}
+    else
+	return 1
+    fi
```

Comment:
Why use a local variable if you aren't manipulating it at all? I would do this:
```suggestion
    lp_container=
    if [[ -n ${SINGULARITY_CONTAINER-${SINGULARITY_NAME-}} ]]; then
        lp_container="Sing"
    elif [[ -e /run/.containerenv ]]; then
        lp_container="Pdmn"
    elif [[ -e /proc/self/cgroup ]]; then
        if grep -q "docker" /proc/self/cgroup ; then
            lp_container="Dokr"
        elif grep -q "lxc" /proc/self/cgroup ; then
            lp_container="LXC"
        fi
    fi

    if [[ -z $lp_container ]] ; then
       return 1
    fi
```

Reply:
In my original implementation, I had the ability to list multiple container contexts (with lines like`container+="LXC"` )

I'll implement your suggestion